sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -161,7 +179,15 @@ case class TransformWithStateExec(
         procTimeIter.flatMap { case rowPair =>
           handleTimerRows(store, rowPair.key, batchTimestampMs.get)
         }
+      case EventTime =>
+        assert(batchTimestampMs.isDefined)
```

Comment:
This is only for the processing time - we should check for the watermark being defined here ?